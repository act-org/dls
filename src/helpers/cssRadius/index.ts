/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import { StandardCSSProperties } from '@mui/system';

import px from '~/helpers/px';

type BorderRadius = Pick<
  StandardCSSProperties,
  | 'borderBottomLeftRadius'
  | 'borderBottomRightRadius'
  | 'borderRadius'
  | 'borderTopLeftRadius'
  | 'borderTopRightRadius'
>;

export const cssRadius = (value: number): BorderRadius => ({
  borderBottomLeftRadius: px(value),
  borderBottomRightRadius: px(value),
  borderRadius: px(value),
  borderTopLeftRadius: px(value),
  borderTopRightRadius: px(value),
});

export default cssRadius;
