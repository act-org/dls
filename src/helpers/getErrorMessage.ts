/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { get } from 'lodash';

export const getErrorMessage = (err: unknown): string => {
  let errorMessage = '';
  const graphQlErrorMessage = get(err, 'networkError.result.errors[0].message');
  const standardErrorMessage = get(err, 'message');

  if (graphQlErrorMessage) {
    errorMessage = graphQlErrorMessage;
  } else if (standardErrorMessage) {
    errorMessage = standardErrorMessage;
  }

  return errorMessage.trim();
};

export default getErrorMessage;
