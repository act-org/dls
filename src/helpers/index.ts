/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

export * from './cssRadius';
export * from './getErrorMessage';
export * from './makeShadow';
export * from './mergeClasses';
export * from './oxfordCommaJoin';
export * from './px';
export * from './pxToNumber';
export * from './search';
export * from './sort';
export * from './styled';
export * from './types';
