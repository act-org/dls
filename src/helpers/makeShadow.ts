/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @prettier
 */

import px from '@actinc/dls/helpers/px';

export const makeShadow = (
  xOffset: number,
  yOffset: number,
  blurRadius: number | undefined,
  spreadRadius: number | undefined,
  color: string,
  inset = false,
): string => {
  return `${inset ? 'inset ' : ''}${px(xOffset)} ${px(yOffset)} ${`${px(
    blurRadius,
  )} `}${`${px(spreadRadius)} `}${color}`;
};

export default makeShadow;
