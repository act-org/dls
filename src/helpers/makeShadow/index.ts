/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import isNumber from 'lodash/isNumber';

import px from '~/helpers/px';

export const makeShadow = (
  xOffset: number,
  yOffset: number,
  blurRadius: number | undefined,
  spreadRadius: number | undefined,
  color: string,
  inset = false,
): string => {
  let str = '';

  if (inset) {
    str = `${str} inset`;
  }

  if (isNumber(xOffset)) {
    str = `${str} ${px(xOffset)}`;
  }

  if (isNumber(yOffset)) {
    str = `${str} ${px(yOffset)}`;
  }

  if (isNumber(blurRadius)) {
    str = `${str} ${px(blurRadius)}`;
  }

  if (isNumber(spreadRadius)) {
    str = `${str} ${px(spreadRadius)}`;
  }

  if (color) {
    str = `${str} ${color}`;
  }

  return str.trim();
};

export default makeShadow;
