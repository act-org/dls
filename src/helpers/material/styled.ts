import { CreateMUIStyled, styled as muiStyled, Theme } from '@mui/material/styles';

export type AsStyledProps<T> = {
  [K in keyof T as `$${string & K}`]: T[K];
};

export const forwardNonTransientProps = (propName: string): boolean => !propName.startsWith('$');

/**
 * Customized styled helper which omits transient props (starting with $)
 * which are used for styling only in styled components.
 * For example you may specify something like <StyledButton $makeItWeird={true} />
 * and this version of styled will not pass $makeItWeird as prop to the HTML in the browser.
 * This is how Styled Components does it: https://styled-components.com/docs/api#transient-props
 * Unfortunately emotion doesn't seem to care: https://github.com/emotion-js/emotion/issues/2193#issuecomment-1178372803
 */
export const styled: CreateMUIStyled<Theme> = ((...args: Parameters<typeof muiStyled>) =>
  muiStyled(args[0], {
    shouldForwardProp: forwardNonTransientProps,
    ...args[1],
  })) as unknown as CreateMUIStyled<Theme>;
