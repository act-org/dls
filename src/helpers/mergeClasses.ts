/**
 * Copyright (c) Encoura, LLC and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import clsx from 'clsx';
import mergeWith from 'lodash/mergeWith';

type Classes = any;

export const mergeClasses = (classes1: Classes, classes2: Classes): Classes =>
  mergeWith(classes1 || {}, classes2 || {}, (className1, className2): string =>
    clsx(className1, className2),
  );

export default mergeClasses;
