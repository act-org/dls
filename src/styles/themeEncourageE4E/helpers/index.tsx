/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

export type ColorIndex =
  | 50
  | 100
  | 200
  | 300
  | 400
  | 500
  | 600
  | 700
  | 800
  | 900
  | string;

interface IBrightnessMap {
  brightness: number;
  colorIndex: ColorIndex;
}

/*
 * Do not modify these values unless UX give us new values,
 * they are used to generate our Figma palette
 */
const brightnessCoefficient: IBrightnessMap[] = [
  {
    brightness: 0.9,
    colorIndex: 50,
  },
  {
    brightness: 0.69,
    colorIndex: 100,
  },
  {
    brightness: 0.54,
    colorIndex: 200,
  },
  {
    brightness: 0.33,
    colorIndex: 300,
  },
  {
    brightness: 0.2,
    colorIndex: 400,
  },
  {
    brightness: 0,
    colorIndex: 500,
  },
  {
    brightness: -0.09,
    colorIndex: 600,
  },
  {
    brightness: -0.29,
    colorIndex: 700,
  },
  {
    brightness: -0.45,
    colorIndex: 800,
  },
  {
    brightness: -0.58,
    colorIndex: 900,
  },
];

export const hexToRGB = (color: string): number[] | undefined => {
  let r = '';
  let g = '';
  let b = '';

  if ((color.length !== 4 && color.length !== 7) || color[0] !== '#')
    return undefined;

  if (color.length === 4) {
    r = `0x${color[1]}${color[1]}`;
    g = `0x${color[2]}${color[2]}`;
    b = `0x${color[3]}${color[3]}`;
  } else if (color.length === 7) {
    r = `0x${color[1]}${color[2]}`;
    g = `0x${color[3]}${color[4]}`;
    b = `0x${color[5]}${color[6]}`;
  }

  return [Number(r), Number(g), Number(b)];
};

export const RGBToHex = (r: number, g: number, b: number): string => {
  let R = r.toString(16);
  let G = g.toString(16);
  let B = b.toString(16);

  if (R.length === 1) R = `0${R}`;
  if (G.length === 1) G = `0${G}`;
  if (B.length === 1) B = `0${B}`;

  return `#${R.slice(0, 2)}${G.slice(0, 2)}${B.slice(0, 2)}`;
};

export const lightenRGB = (
  color: string,
  percentage: number,
): string | undefined => {
  const RGBColor = hexToRGB(color);

  if (!RGBColor) return undefined;

  if (percentage === 0) return color;

  if (percentage > 0) {
    return RGBToHex(
      Math.round(RGBColor[0] + ((255 - RGBColor[0]) * percentage) / 100),
      Math.round(RGBColor[1] + ((255 - RGBColor[1]) * percentage) / 100),
      Math.round(RGBColor[2] + ((255 - RGBColor[2]) * percentage) / 100),
    );
  }

  return RGBToHex(
    Math.round(RGBColor[0] * (1 + percentage / 100)),
    Math.round(RGBColor[1] * (1 + percentage / 100)),
    Math.round(RGBColor[2] * (1 + percentage / 100)),
  );
};

export const figmaColorGenerator = (
  color: string,
): Record<ColorIndex, string> => {
  const colorPalette = {} as unknown as Record<ColorIndex, string>;
  brightnessCoefficient.forEach(({ brightness, colorIndex }) => {
    const generatedColor = lightenRGB(color, brightness * 100);
    if (generatedColor) colorPalette[colorIndex] = generatedColor;
  });
  return colorPalette;
};

export const generatePaletteColors = <
  T extends { colorName: U; colorHEX: string },
  U extends string,
>(
  colors: T[],
): Record<U, Record<ColorIndex, string>> => {
  const colorPalette = {} as unknown as Record<U, Record<ColorIndex, string>>;
  colors.forEach(color => {
    colorPalette[color.colorName] = figmaColorGenerator(color.colorHEX);
  });
  return colorPalette;
};
