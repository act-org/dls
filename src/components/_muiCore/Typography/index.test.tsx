/**
 * Copyright (c) ACT, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { Typography } from '@mui/material';

import { standard } from '~/helpers/test';

describe('Typography', () => {
  const Component = <Typography variant="h1">Heading Title</Typography>;
  standard(Component);
});
