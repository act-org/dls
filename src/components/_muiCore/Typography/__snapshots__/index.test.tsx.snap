// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography ACT theme matches the snapshot 1`] = `
<div>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-1b93ly8-MuiTypography-root"
  >
    Heading Title
  </h1>
</div>
`;

exports[`Typography ACT_ET theme matches the snapshot 1`] = `
<div>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-12eg8o5-MuiTypography-root"
  >
    Heading Title
  </h1>
</div>
`;

exports[`Typography ENCOURA theme matches the snapshot 1`] = `
<div>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-1f28kiy-MuiTypography-root"
  >
    Heading Title
  </h1>
</div>
`;

exports[`Typography ENCOURA_CLASSIC theme matches the snapshot 1`] = `
<div>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-1f28kiy-MuiTypography-root"
  >
    Heading Title
  </h1>
</div>
`;

exports[`Typography ENCOURAGE theme matches the snapshot 1`] = `
<div>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-12tk7s5-MuiTypography-root"
  >
    Heading Title
  </h1>
</div>
`;

exports[`Typography ENCOURAGE_E4E theme matches the snapshot 1`] = `
<div>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-11dypxl-MuiTypography-root"
  >
    Heading Title
  </h1>
</div>
`;
